From 2843380fea2d2342f907a0f615e7baa5ea5d47d9 Mon Sep 17 00:00:00 2001
From: kongfl888 K <kongfl888@outlook.com>
Date: Wed, 9 Sep 2020 21:04:16 +0800
Subject: [PATCH] rockchip:nanopi-r2s improve booting on SD Card.

It will improve the problem that in some SD cards are failing to boot up.

And it may useful for Class 4 SD Cards.

Issues:
  When booting on some SD cards an error message appears as:
   "spl: mmc init failed with error: -95"
   "mmc0: error -110 whilst initialising SD card"

Solutions:
  Add regulator-boot-on parameters to vcc_sd.

Signed-off-by: kongfl888 K <kongfl888@outlook.com>
---
 arch/arm64/boot/dts/rockchip/rk3328-nanopi-r2-common.dtsi | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm64/boot/dts/rockchip/rk3328-nanopi-r2-common.dtsi b/arch/arm64/boot/dts/rockchip/rk3328-nanopi-r2-common.dtsi
index e2c06db..df7bb2a 100644
--- a/arch/arm64/boot/dts/rockchip/rk3328-nanopi-r2-common.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3328-nanopi-r2-common.dtsi
@@ -92,6 +92,7 @@
 		regulator-name = "vcc_sd";
 		regulator-min-microvolt = <3300000>;
 		regulator-max-microvolt = <3300000>;
+		regulator-boot-on;
 		vin-supply = <&vcc_io>;
 	};
 
-- 
2.14.1.windows.1

